### PR TITLE
chore/ag/finalize release kadena client

### PR DESCRIPTION
**Pull request by alber70g**
- chore(client): finalize the api and add ts-doc
- chore(generator): use the new ICap instead of ICapabilityItem
- refactor(tools): use contractParser insteda of StringContractDefinition

### CODE REVIEW

**Review by javadkh2 — APPROVED** [review]
all good!

File: packages/libs/client/src/composePactCommand/utils/addKeyset.ts [hunk 1]
```diff
@@ -0,0 +1,12 @@
+import { addData, IAddKeyset } from './addData';
+
+/**
+ * Helper to add keyset to the data property for {@link IPactCommand.payload}
+ *
+ * @public
+ */
+export const addKeyset: IAddKeyset = (
+  name: string,
+  pred: string,
```
Comment (webpro) [inline packages/libs/client/src/composePactCommand/utils/addKeyset.ts:10]: Nitpick: not common to shorten `pred`

File: packages/libs/client/src/composePactCommand/utils/patchCommand.ts [hunk 2]
```diff
@@ -47,6 +50,16 @@ export const mergePayload = (
   throw new Error('PAYLOAD_NOT_MERGEABLE');
 };
 
+/**
+ * Merge a partial command on top of the command
+ *
+ * @remarks
```
Comment (webpro) [inline packages/libs/client/src/composePactCommand/utils/patchCommand.ts:56]: Nitpick: unknown JSDoc tag?

File: packages/libs/client/src/createTransactionBuilder/createTransactionBuilder.ts [hunk 3]
```diff
@@ -80,7 +77,11 @@ interface IAddKeyset<TCommand> {
   ): IBuilder<TCommand>;
 }
 /**
- * @alpha
+ * The interface of the return value `Pact.builder.execution` or `Pact.builder.continuation`
+ *
+ * @remarks
+ * See {@link Pact}
```
Comment (webpro) [inline packages/libs/client/src/createTransactionBuilder/createTransactionBuilder.ts:83]: And `@see` is a known JSDoc tag

File: packages/libs/client/src/signing/walletconnect/walletConnectTypes.ts [hunk 4]
```diff
@@ -33,13 +37,13 @@ export interface IWalletConnectAccount {
...
-  walletConnectChainId: TWalletConnectChainId;
-  network: IPactCommand['networkId']; // Kadena network (mainnet, testnet, devnet)
-  account: string; // Kadena account
-  chainId: ChainwebChainId; // Kadena ChainId
-}
-
-export type TSigningType = 'sign' | 'quicksign';
-
+/**
+ * The Blockchain that's used WalletConnect context
```
Comment (webpro) [inline packages/libs/client/src/signing/walletconnect/walletConnectTypes.ts:41]:
```suggestion
 * The Blockchain that's used in a WalletConnect context
```

File: packages/libs/client/src/utils/tests/parseAsPactValue.test.ts [hunk 5]
```diff
@@ -43,9 +43,11 @@ describe('parseType', () => {
     expect(parseAsPactValue(() => 'test')).toEqual('test');
```
Comment (webpro) [inline packages/libs/client/src/utils/tests/parseAsPactValue.test.ts:43]: Not part of this PR, but is executing those functions a potential security concern?

File: packages/libs/pactjs-generator/src/contract/generation/tests/__snapshots__/generator.test.ts.snap [hunk 6]
```diff
@@ -203,11 +196,10 @@ declare module '@kadena/client' {
 
 exports[`generateDts uses the property type if there is no mapped value for that 1`] = `
 "
-import type { ICapabilityItem } from '@kadena/client';
-import type { IPactDecimal, IPactInt, PactReference } from '@kadena/types';
+import type { IPactDecimal, IPactInt, PactReference, ICap } from '@kadena/types';
 
 interface ICapability_Coin_GAS {
-  (name: 'coin.GAS'): ICapabilityItem;
+  (name: 'coin.GAS'): ICap;
```
Comment (webpro) [inline packages/libs/pactjs-generator/src/contract/generation/tests/__snapshots__/generator.test.ts.snap:202]: Nitpick: seeing all these renames making a lot of sense, except this one is the only shortened.

**Review by webpro — APPROVED** [review]
LGTM (without running anything)

File: packages/libs/client/src/composePactCommand/utils/patchCommand.ts (hunk 2 above)
Comment (alber70g) [reply packages/libs/client/src/composePactCommand/utils/patchCommand.ts:56]: It's TSDoc. Amongst others, JSDoc includes types which are redundant in TSDoc

File: packages/libs/client/src/composePactCommand/utils/addKeyset.ts (hunk 1 above)
Comment (alber70g) [reply packages/libs/client/src/composePactCommand/utils/addKeyset.ts:10]: We used this to keep familiarity with a [Keyset definition](https://pact-language.readthedocs.io/en/stable/pact-reference.html#keyset-definition-1)

File: packages/libs/client/src/createTransactionBuilder/createTransactionBuilder.ts (hunk 3 above)
Comment (alber70g) [reply packages/libs/client/src/createTransactionBuilder/createTransactionBuilder.ts:83]: Modified, will update others as well

File: packages/libs/client/src/signing/walletconnect/walletConnectTypes.ts (hunk 4 above)
Comment (alber70g) [reply packages/libs/client/src/signing/walletconnect/walletConnectTypes.ts:41]: :+1:

File: packages/libs/client/src/utils/tests/parseAsPactValue.test.ts (hunk 5 above)
Comment (alber70g) [reply packages/libs/client/src/utils/tests/parseAsPactValue.test.ts:43]: WDYM? As a potential injection in the Pact code?

That should be covered by proper xss protection on the hosting website, I guess?

File: packages/libs/pactjs-generator/src/contract/generation/tests/__snapshots__/generator.test.ts.snap (hunk 6 above)
Comment (alber70g) [reply packages/libs/pactjs-generator/src/contract/generation/tests/__snapshots__/generator.test.ts.snap:202]: Yes, we moved away from just wrapping the `ICap` from `@kadena/types` as `ICapabilityItem` and chose to use the `ICap` directly
